persistence/src/test/java/edu/unc/lib/dl/persist/services/destroy/DestroyObjectsJobIT.java

```diff
@@ -0,0 +1,268 @@
+        FolderObject folderObj = repoObjLoader.getFolderObject(folderObjPid);
+        assertTrue(fileObj.getModel().contains(fileObj.getResource(), RDF.type, Cdr.Tombstone));
+        assertTrue(workObj.getModel().contains(workObj.getResource(), RDF.type, Cdr.Tombstone));
+        assertTrue(folderObj.getModel().contains(folderObj.getResource(), RDF.type, Cdr.Tombstone));
+
+        PremisEventObject event = repoObjLoader.getPremisEventObject(folderObj.getPremisLog().listEvents().get(0));
+        assertTrue(event.getResource().hasProperty(Premis.hasEventType, Premis.Deletion));
+        assertTrue(event.getResource().hasProperty(Premis.hasEventDetail,
+                "Item deleted from repository and replaced by tombstone"));
+    }
+
+    @Test
+    public void destroyObjectsInDifferentTreesTest() {
+        initializeJob(objsToDestroy);
+
+        job.run();
+
+        PID folderObj2Pid = objsToDestroy.get(3);
+        PID fileObjPid = objsToDestroy.get(2);
+        PID workObjPid = objsToDestroy.get(1);
+        PID folderObjPid = objsToDestroy.get(0);
+        verify(spyProxyService).destroyProxy(folderObjPid);
+        verify(spyProxyService).destroyProxy(folderObj2Pid);
+
+        FileObject fileObj = repoObjLoader.getFileObject(fileObjPid);
+        WorkObject workObj = repoObjLoader.getWorkObject(workObjPid);
+        FolderObject folderObj = repoObjLoader.getFolderObject(folderObjPid);
+        FolderObject folderObj2 = repoObjLoader.getFolderObject(folderObj2Pid);
+        assertTrue(fileObj.getModel().contains(fileObj.getResource(), RDF.type, Cdr.Tombstone));
+        assertTrue(workObj.getModel().contains(workObj.getResource(), RDF.type, Cdr.Tombstone));
+        assertTrue(folderObj.getModel().contains(folderObj.getResource(), RDF.type, Cdr.Tombstone));
+        assertTrue(folderObj2.getModel().contains(folderObj2.getResource(), RDF.type, Cdr.Tombstone));
+
+        PremisEventObject event = repoObjLoader.getPremisEventObject(folderObj.getPremisLog().listEvents().get(0));
+        assertTrue(event.getResource().hasProperty(Premis.hasEventType, Premis.Deletion));
+        assertTrue(event.getResource().hasProperty(Premis.hasEventDetail,
+                "Item deleted from repository and replaced by tombstone"));
+
+        PremisEventObject event2 = repoObjLoader.getPremisEventObject(folderObj2.getPremisLog().listEvents().get(0));
+        assertTrue(event.getResource().hasProperty(Premis.hasEventType, Premis.Deletion));
+        assertTrue(event.getResource().hasProperty(Premis.hasEventDetail,
+                "Item deleted from repository and replaced by tombstone"));
+    }
+
+    @Test
+    public void destroyFolderTest() {
+        PID folderObjPid = objsToDestroy.get(0);
+        initializeJob(Arrays.asList(folderObjPid));
+        FolderObject folderObj = repoObjLoader.getFolderObject(folderObjPid);
+        WorkObject workObj = repoObjLoader.getWorkObject(folderObj.getMembers().get(0).getPid());
```

Comment:
Why not cast the member instead of re-retrieving it?
`WorkObject workObj = (WorkObject) folderObj.getMembers().get(0);`

Reply:
okie dokie